src/main/java/io/cdap/delta/datastream/DatastreamEventReader.java

```diff
@@ -347,14 +353,19 @@ emit DML event  (table.pos = current pos , pos++)
               .fields(Storage.BlobField.NAME, Storage.BlobField.TIME_CREATED, Storage.BlobField.SIZE));
 
             scanEvents(blobs, tableName, srcTable, true, path != null);
-            // TODO use Datastream API to determine whether dump is done, below is just a workaround
-            // if new dump file found  then that means dump hasn't been completed
-            if (getPath(tableName) == null || !getPath(tableName).equals(path)) {
+
+            // checks whether backfill is done for the table
+            if (!tableBackFillStatuses.containsKey(srcTable.getTable())) {
```

Comment:
we already have `getDumped(tableName)`  could you combine them ?

Reply:
mainly , now we have two source of truth for whether a table is done with backfill

one stored in the state
and the other are got from datastream API